### DISCUSSION TITLE
Why doesn't the AR button appear in Chrome on iPhone?

### DISCUSSION THREAD

**Discussion by MilosNikolic** [google/model-viewer#1871] [Q&A]
### Description
I started playing with <model-viewer> and I noticed that AR button is not showing in Chrome on iPhone.
Phone i was using is iPhone 6s, with iOS 12.4
Here is the glitch project i used for testing:
https://glitch.com/edit/#!/midnight-friday

Interestingly, it works fine in Safari on iPhone, but not in Chrome. 

#### Live Demo

https://glitch.com/edit/#!/midnight-friday

### Browser Affected

- [x] Chrome
- [ ] Edge
- [ ] Firefox
- [ ] Helios
- [ ] IE
- [ ] Safari

### OS

- [ ] Android
- [x] iOS 12.4
- [ ] Linux
- [ ] MacOS
- [ ] Windows

### Versions

- model-viewer: v0.6.0

**Comment by makc** [comment]
related: #758 ?

**Comment by cwervo** [comment]
Hey @MilosNikolic I forked your Glitch demo into [a CodePen example](https://codepen.io/cwervo/pen/BaBErLy) and added this line:

```html
quick-look-browsers="safari chrome"
```

This tells the `model-viewer` component to make this available on both Chrome & Safari on iOS & works on my iOS device:

![IMG_0445 copy](https://user-images.githubusercontent.com/5826638/65739418-2a013200-e0b3-11e9-9013-575b49a84eb5.PNG)

Does that solve the issue for you?

**Comment by MilosNikolic** [comment]
@cwervo - Hi Andrés, that did work indeed, icon showed up, but the model is still not loading in AR, this is using both your remixed code and when I add it, which means its essentially not a solution.

Thanks for checking it tho!

**Comment by cwervo** [comment]
@MilosNikolic Do you get an error, a gray object panel showing something like "object cannot be loaded", or just no feedback at all?

**Comment by cdata** [comment]
Hi all, due to an inconsistency in how iOS Chrome handles `.usdz` files, you have to opt-in to AR in iOS Chrome. Please refer to the `quick-look-browsers` attribute in our documentation.

Be advised that until recently, iOS Chrome _does not activate Quick Look unless you set the correct `Content-Type` header when serving the `.usdz` file_. If the header is not set correctly, iOS Chrome just downloads the file and it is a pretty lame experience.

An example of opting in looks like this:

```html
<model-viewer src="..." ios-src="any.usdz" quick-look-browsers="safari chrome">
</model-viewer>
```

Please also refer to this related issue: https://github.com/GoogleWebComponents/model-viewer/issues/534

**Comment by cdata** [comment]
Also worth mentioning that recent versions of Chrome should behave the same as Safari even if the header is omitted, but for now we are making it opt-in until the inconsistency is really far in the rear-view mirror.

**Comment by cdata** [comment]
If there are no further issues, I'm going to re-categorize this issue as a question.

**Comment by pixelrgb028** [comment]
I have 4 products to choose from, and am able to update the path of my .glb product models with an onClick event that updates the 'src' attribute. But when I try to update the 'ios-src' attribute to view the usdz version of the model in QuickLook on an iPhone 6S, I end up with "Object could not be opened". 

However, if I link the path of the 'ios-src' to a single usdz directly, the browser tiggers QuickLook and it works. 

After checking the inspector I find the following warning------------------
DevTools failed to load SourceMap: Could not load content for https://unpkg.com/@google/model-viewer@1.0.0/dist/model-viewer.js.map: HTTP error: status code 404, net::ERR_HTTP_RESPONSE_CODE_FAILURE

*Console.log('ios-src') renders undefined; is this attribute available to modifications at runtime? I really appreciate any advise you have on this issue..

**Comment by elalish** [comment]
@pixelrgb028 It's possible this is the same error as #1286, but for quicklook instead. I think I fixed them both; can you verify if this repros for you on master?

**Comment by elalish** [comment]
@pixelrgb028 And just checking since someone else had this trouble; the attribute name is `ios-src`, but in JS, the member is called `iosSrc`, as this is the traditional naming convention on the web.
